CMakeLists.txt

```diff
@@ -101,10 +101,10 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Os -Wno-sized-deallocation -fno-sized-d
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DDEBUG -DSANITIZE_PTRS")
 add_link_options(--sysroot ${CMAKE_BINARY_DIR}/Root)
 
-if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wconsumed")
-elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" OR "${CMAKE_SYSTEM_NAME}" STREQUAL "FreeBSD")
```

Comment:
This seems weird. What is the `CMAKE_CXX_COMPILER_ID` on your system?

Reply:
It would actually resolve to Clang on my system.

Reply:
Something here doesn't add up. Your compiler is GCC but CMake identifies it as Clang?
Can anyone else with FreeBSD confirm this behavior?

Reply:
Actually, checking again I do have clang, so the detection is correct. However, it seems to dislike `-Wconsumed` and requires `-Wno-expansion-to-defined` to compile, favoring the GNU path.

If anyone has an ability to test it out on FreeBSD, I'd be glad.

Reply:
Okay, found out the reason. When CMake generates the project files, it detects my clang and correctly sets up the flags based on the rules.

However, since `-Wconsumed` is a flag served for clang and we compile the code using the `i686-pc-serenity-` cross compiler, it basically freaks out.

Since FreeBSD uses clang by default, it explains why I got into this situation at the first place. :)